gluoncv/model_zoo/center_net/deconv_resnet.py

```diff
@@ -0,0 +1,266 @@
+
+
+class BilinearUpSample(mx.init.Initializer):
+    """Initializes weights as bilinear upsampling kernel.
+
+    Example
+    -------
+    >>> # Given 'module', an instance of 'mxnet.module.Module',
+    initialize weights to bilinear upsample...
+    >>> init = mx.initializer.BilinearUpSample()
+    >>> module.init_params(init)
+    >>> for dictionary in module.get_params():
+    ...     for key in dictionary:
+    ...         print(key)
+    ...         print(dictionary[key].asnumpy())
+    ...
+    fullyconnected0_weight
+    [[ 0.  0.  0.]]
+    """
+    def __init__(self):
+        super(BilinearUpSample, self).__init__()
+
+    def _init_weight(self, _, arr):
+        mx.nd.random.normal(0, 0.01, arr.shape, out=arr)
+        f = math.ceil(arr.shape[2] / 2)
+        c = (2 * f - 1 - f % 2) / (2. * f)
+        for i in range(arr.shape[2]):
+            for j in range(arr.shape[3]):
+                arr[0, 0, i, j] = \
+                    (1 - math.fabs(i / f - c)) * (1 - math.fabs(j / f - c))
+        for c in range(1, arr.shape[0]):
+            arr[c, 0, :, :] = arr[0, 0, :, :]
+
+class DeconvResnet(nn.HybridBlock):
+    """Deconvolutional ResNet.
+
+    Parameters
+    ----------
+    base_network : str
+        Name of the base feature extraction network.
+    deconv_filters : list of int
+        Number of filters for deconv layers.
+    deconv_kernels : list of int
+        Kernel sizes for deconv layers.
+    pretrained_base : bool
+        Whether load pretrained base network.
+    norm_layer : mxnet.gluon.nn.HybridBlock
+        Type of Norm layers, can be BatchNorm, SyncBatchNorm, GroupNorm, etc.
+    norm_kwargs : dict
+        Additional kwargs for `norm_layer`.
+    use_dcnv2 : bool
+        If true, will use DCNv2 layers in upsampling blocks
+
+    """
+    def __init__(self, base_network='resnet18_v1b',
+                 deconv_filters=(256, 128, 64), deconv_kernels=(4, 4, 4),
+                 pretrained_base=True, norm_layer=nn.BatchNorm, norm_kwargs=None,
+                 use_dcnv2=False, **kwargs):
+        super(DeconvResnet, self).__init__(**kwargs)
+        assert 'resnet' in base_network
+        net = get_model(base_network, pretrained=pretrained_base)
+        self._norm_layer = norm_layer
+        self._norm_kwargs = norm_kwargs if norm_kwargs is not None else {}
+        self._use_dcnv2 = use_dcnv2
+        if 'v1b' in base_network:
+            feat = nn.HybridSequential()
+            feat.add(*[net.conv1,
+                       net.bn1,
+                       net.relu,
+                       net.maxpool,
+                       net.layer1,
+                       net.layer2,
+                       net.layer3,
+                       net.layer4])
+            self.base_network = feat
+        else:
+            raise NotImplementedError()
+        with self.name_scope():
+            self.deconv = self._make_deconv_layer(deconv_filters, deconv_kernels)
```

Comment:
in pose estimation pixel shuffle achieves similar acc with much less computation comparing to deconv, will we consider that?

Reply:
There's a plan for developing a fast version so pixel shuffle could be a better choice